api/app/Http/Controllers/BaseController.php

```diff
@@ -153,68 +157,4 @@ public function deleteMany(Request $request)
 
         return response(null, 204);
     }
```

Comment:
OK I've just had a discussion with Andre about this, and we both agree that we think that the repository should have the sole responsibility of storing and retrieving models. The controller should basically puppet the validator, authentication, permission, models and repository services, getting the responses it needs and shipping them back to the user via a transformer.

While there shouldn't be too much business logic in the controller I think it is a mistake to delegate the linking of entities to the repository - that should take place in the controller. That way repositories can be solely responsible for storage and retrieval, and if say we switch to mongo for example it should be trivial to update as the repositories are not coupled to the business logic.

@Redjik I do like the methods that you laid out there, I think it makes sense to have the pattern like that. Ideally if we can continue to use the eloquent structure it would be great as it is easier to refer to their documentation i.e. http://laravel.com/docs/5.1/eloquent-relationships#inserting-related-models